File: feel/model/friend/friend_request.go
package friend

import (
	_ "github.com/go-playground/validator/v10"
)

type BuddyRequest struct {
	Id       string `db:"id"`
	UserId   string `db:"userId"`
	FromId   string `json:"fromId" validate:"required" db:"fromId"`
	ToId     string `json:"toId" validate:"required" db:"toId"`
	CreateAt string `json:"createAt" validate:"required" db:"createAt"`
	Relation string `json:"relation" validate:"required" db:"relation"`
	Status   string `json:"status" validate:"required" db:"status"`
}

const (
	Request    string = "request_friend"
	WaitAccept string = "wait_accept_friend"
	Accept     string = "accept"
	Bloc       string = "bloc_friend"
	IsFriend   string = "make_friend"
	UnFriend   string = "un_friend"
)

func BuildFriend() *BuddyRequest {
	return &BuddyRequest{
		Id:       "",
		FromId:   "",
		ToId:     "",
		CreateAt: "",
		Relation: "",
		Status:   "",
	}
}

File: lib/data/models/friend/friend_request.dart
class FriendRequest {
  final String fromId;
  final String toId;
  final String createAt;
  final String relation;
  final String status;

  FriendRequest(
      {this.fromId = "",
      this.toId = "",
      this.createAt = "",
      this.relation = "",
      this.status = ""});

  factory FriendRequest.fromJson(Map<String,String> json) => FriendRequest(
    fromId: json["fromId"].toString(),
    toId: json["toId"].toString(),
    createAt: json["createAt"].toString(),
    relation: json["relation"].toString(),
    status: json["status"].toString()
  );

  Map<String, String> toJson() => Map.of({
        "fromId": fromId,
        "toId": toId,
        "createAt": createAt,
        "relation": relation,
        "status": status
      });
}
